lib/rex/post/mssql/ui/console/command_dispatcher/client.rb

```diff
@@ -13,132 +10,18 @@ module Ui
-            print_line 'To exit, type `exit`, `quit`, `end` or `stop`.'
-            print_line
-          end
-
-          # @param [Array] args
-          # @return [Object]
-          def cmd_shell(*args)
-            cmd_shell_help && return if help_args?(args)
-
-            prompt_proc_before = ::Reline.prompt_proc
-
-            ::Reline.prompt_proc = proc { |line_buffer| line_buffer.each_with_index.map { |_line, i| i > 0 ? 'SQL *> ' : 'SQL >> ' } }
-
-            stop_words = %w[stop s exit e end quit q].freeze
-
-            finished = false
-            loop do
-              begin
-                raw_query = ::Reline.readmultiline('SQL >> ', use_history = true) do |multiline_input|
-                  finished = stop_words.include?(multiline_input.split.last)
-                  finished || (multiline_input.split.last && !multiline_input.split.last.end_with?('\\'))
-                end
-              rescue ::Interrupt
-                finished = true
-              ensure
-                ::Reline.prompt_proc = prompt_proc_before
-              end
-
-              if finished
-                print_status 'Exiting Shell mode.'
-                return
-              end
-
-              formatted_query = raw_query.split.map { |word| word.chomp('\\') }.reject(&:empty?).compact.join(' ')
-
-              print_status "Running SQL Command: '#{formatted_query}'"
-              cmd_query(formatted_query)
-            end
-          end
-
-          # @return [Object]
           def cmd_query_help
-            print_line 'Usage: query'
-            print_line
-            print_line 'Run a raw SQL query on the target.'
-            print_line 'Examples:'
-            print_line
-            print_line '    query select @@version;'
-            print_line '    query select user_name();'
-            print_line '    query select name from master.dbo.sysdatabases;'
-            print_line
-          end
-
-          # @param [Array] result The result of an SQL query to format.
-          def format_result(result)
-            columns = ['#']
-
-            unless result.is_a?(Array)
-              result.fields.each { |field| columns.append(field.name) }
-
-              ::Rex::Text::Table.new(
-                'Header' => 'Query Result',
-                'Indent' => 4,
-                'Columns' => columns,
-                'Rows' => result.map.each.with_index { |row, i| [i, row].flatten }
-              )
-            end
-          end
-
-          # @param [Array] args SQL query
-          # @return [Object]
-          def cmd_query(*args)
-            if help_args?(args)
-              cmd_query_help
-              return
-            end
-
-            query = args.join(' ').to_s
-            client.mssql_query(query, true) || []
+            examples = ['select @@version;', 'select user_name();', 'select name from master.dbo.sysdatabases;']
+            super(examples: examples)
```

Comment:
This seems like a code smell, having `cmd_query_help` take an array of strings, and then now you're calling `super` with your own API. This approach breaks composability and element of least surprise IMO

Probably an easy solution is just:

```suggestion
            some_other_method(examples: examples)
```